<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>681</width>
    <height>597</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>130</y>
      <width>656</width>
      <height>258</height>
     </rect>
    </property>
    <property name="title">
     <string>预编码数据序列化方式读写测试</string>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>qint8(-128~127)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_8">
         <property name="minimum">
          <number>-128</number>
         </property>
         <property name="maximum">
          <number>127</number>
         </property>
         <property name="value">
          <number>85</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_14">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_r8">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_8"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>quint(0~255)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_u8">
         <property name="maximum">
          <number>255</number>
         </property>
         <property name="value">
          <number>236</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_u8">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_ru8">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_u8"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>qint16(-32768~32767)l</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_16">
         <property name="minimum">
          <number>-32768</number>
         </property>
         <property name="maximum">
          <number>32767</number>
         </property>
         <property name="value">
          <number>-1522</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_16">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_r16">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_16"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>quint16(0~65535)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_u16">
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>54862</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_u16">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_ru16">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_u16"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>qint32</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_32">
         <property name="minimum">
          <number>-65535</number>
         </property>
         <property name="maximum">
          <number>65535</number>
         </property>
         <property name="value">
          <number>-3546</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_32">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_r32">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_32"/>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>qint64</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="spinBox_64">
         <property name="minimum">
          <number>-1000000</number>
         </property>
         <property name="maximum">
          <number>10000000</number>
         </property>
         <property name="value">
          <number>2564851</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_64">
         <property name="text">
          <string>写入</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_r64">
         <property name="text">
          <string>读出</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineEdit_64"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>641</width>
      <height>30</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="pushButton_testfile">
       <property name="text">
        <string>测试用文件</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="lineEdit"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>50</y>
      <width>641</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>字节序</string>
       </property>
       <widget class="QRadioButton" name="radioButton_Little">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>118</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>LittleEndian</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_big">
        <property name="geometry">
         <rect>
          <x>130</x>
          <y>30</y>
          <width>118</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>BigEndian</string>
        </property>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>浮点数精度</string>
       </property>
       <widget class="QRadioButton" name="radioButton_single">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>30</y>
          <width>141</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>singleprecision</string>
        </property>
       </widget>
       <widget class="QRadioButton" name="radioButton_double">
        <property name="geometry">
         <rect>
          <x>160</x>
          <y>30</y>
          <width>131</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>DoublePrecision</string>
        </property>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="font">
    <font>
     <pointsize>14</pointsize>
    </font>
   </property>
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionclear_edit"/>
   <addaction name="separator"/>
   <addaction name="actionwrite_file"/>
   <addaction name="separator"/>
   <addaction name="actionget_file"/>
   <addaction name="separator"/>
   <addaction name="actionquit"/>
  </widget>
  <action name="actionclear_edit">
   <property name="text">
    <string>clear_edit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionwrite_file">
   <property name="text">
    <string>write_file</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionget_file">
   <property name="text">
    <string>get_file</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
  <action name="actionquit">
   <property name="text">
    <string>quit</string>
   </property>
   <property name="menuRole">
    <enum>QAction::NoRole</enum>
   </property>
  </action>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>actionquit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>340</x>
     <y>298</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
